### PR TITLE
Manage authentication

### PR DESCRIPTION
**Pull request by lemald**
#### Summary of changes
**Asana Ticket:** [Manage authentication on the web app](https://app.asana.com/0/0/1122041752019628/f)

[Please include a brief description of what was changed]

#### Reviewer Checklist
- [ ] Meets ticket's acceptance criteria
- [ ] Any new or changed functions have typespecs
- [ ] Tests were added for any new functionality (don't just rely on Codecov)
- [ ] This branch was deployed to the staging environment and is currently running with no unexpected increase in warnings, and no errors or crashes (compare on Splunk: [staging](https://mbta.splunkcloud.com/en-US/app/search/search?q=search%20index%3Dsigns-ui-dev%20%22%5Berror%5D%22%20OR%20%22%5Bwarn%5D%22%20OR%20%22CRASH%22&display.page.search.mode=verbose&dispatch.sample_ratio=1&earliest=-4h%40m&latest=now&sid=1545840769.3874970) vs. [prod](https://mbta.splunkcloud.com/en-US/app/search/search?q=search%20index%3Dsigns-ui-prod%20%22%5Berror%5D%22%20OR%20%22%5Bwarn%5D%22%20OR%20%22CRASH%22&display.page.search.mode=verbose&dispatch.sample_ratio=1&earliest=-4h%40m&latest=now&sid=1545840745.3874956))

### CODE REVIEW

File: lib/signs_ui_web/router.ex [hunk 1]
```diff
@@ -68,4 +64,18 @@ defmodule SignsUiWeb.Router do
...
+
+  @spec cognito_auth(Plug.Conn.t(), Plug.opts()) :: Plug.Conn.t()
+  defp cognito_auth(conn, _) do
+    conn = fetch_session(conn)
+    login_expiration = get_session(conn, :login_expiration)
+
+    current_time = DateTime.utc_now() |> DateTime.to_unix()
+
+    if is_nil(login_expiration) or current_time >= login_expiration do
+      conn |> redirect(to: "/cognito/new") |> halt()
```
Comment (losvedir) [inline lib/signs_ui_web/router.ex]: I think `/cognito/new` has a path helper function. Try `mix phx.routes` to see, and then I think you might need to do something like `Router.foo_path()` to access it. Although, come to think of it you might not be able to from _within_ the router...

Comment (lemald) [reply lib/signs_ui_web/router.ex]: Hmm, so I'm trying `Router.cognito_path` and not getting any luck. I've tried it both without arguments, and also with `:new`. It's worth noting that `mix phx.routes` reports two different entries for `cognito_path` (one for `/cognito` and one for `/cognito/new`), but both `cognito_path/0` and `cognito_path/1` are apparently undefined.

Comment (losvedir) [reply lib/signs_ui_web/router.ex]: The path helper is typically at least `/2` (or more if takes an ID). The first argument is either a `conn` if you have one, or the module name for the Endpoint. So try `SignsUiWeb.Router.Helpers.cognito_path(conn, :new)`.

Comment (lemald) [reply lib/signs_ui_web/router.ex]: Ah, thanks. I managed to get it working.

**Pull request merged**